### PR TITLE
Release v7.1

### PR DESCRIPTION
**Pull request by chrisjsewell**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by chrisjsewell** [comment]
@conda-forge-admin, please rerender

**Comment by chrisjsewell** [comment]
@jan-janssen @wolfv 

```
RuntimeError: Solver could not find solution.Mamba failed to solve:
 - cmake >=2.8.12
 - openmpi-mpifort
 - wget
 - m4
 - pkg-config
 - gfortran_osx-64 11.*
 - clang_osx-64 14.*
 - make
 - llvm-openmp 14.*

with channels:

The reported errors are:
- Encountered problems while solving:
-   - nothing provides openmpi 4.0.3 hed52333_0 needed by openmpi-mpifort-4.0.3-he951c17_0
- 
```

I didn't change any of the requirements here (I don't think they need changing?) so suprised it now won't allow the solve

Is this because 7.0 was, presumably, built using the non-mamba solver, which didn't detect this incompatibility  🤷

**Comment by ltalirz** [comment]
At least it's the first time I'm seeing reports from the mamba solver

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/QEF/q-e/archive/{{ name }}-{{ version }}.tar.gz
-  sha256: 85beceb1aaa1678a49e774c085866d4612d9d64108e0ac49b23152c8622880ee
-  patches:
-   # drop following patch for next version after 7.0
```
Comment (chrisjsewell) [inline recipe/meta.yaml:12]: I assume this was ok to remove?

**Comment by chrisjsewell** [comment]
> I didn't change any of the requirements here (I don't think they need changing?) so suprised it now won't allow the solve

Ah no, I think its due to the changes in `[.ci_support/osx_64_.yaml](https://github.com/conda-forge/qe-feedstock/pull/32/files#diff-92d4214334555338c6e098e1462e4dec3ca2032a9f2973bf5558d6b8c00503ce)`

Does this mean we need to change the build requirements in some way? (perhaps `llvm-openmp` ?)

File: recipe/meta.yaml (hunk 1 above)
Comment (ltalirz) [reply recipe/meta.yaml:12]: yes

**Comment by ltalirz** [comment]
possible

**Comment by chrisjsewell** [comment]
Ok, osk-64 is now building (I guess dependencies were updated)

BUT, now its failing because `make install` does not appear to be finding all the executables 🤷 

If you look at the build logs, it appears to be creating all the executables, but then just not installing them, e.g.

```python
In [1]: sorted(re.findall(r"Linking Fortran executable \.\./bin/(.*\.x)", text))
Out[1]: 
['ZG.x',
 'abcoeff_to_eps.x',
 'all_currents.x',
 'alpha2f.x',
 'average.x',
 'bands.x',
 'bands_unfold.x',
 'bse_main.x',
 'casino2upf.x',
 'cell2ibrav.x',
 'cp.x',
 'cppp.x',
 'disca.x',
 'dos.x',
 'dos_sp.x',
 'dvscf_q2r.x',
 'dynmat.x',
 'ef.x',
 'epa.x',
 'epsilon.x',
 'epsilon_Gaus.x',
 'epw.x',
 'ev.x',
 'fd.x',
 'fd_ef.x',
 'fd_ifc.x',
 'fermi_int_0.x',
 'fermi_int_1.x',
 'fermi_proj.x',
 'fermi_velocity.x',
 'fqha.x',
 'fs.x',
 'graph.x',
 'gww.x',
 'gww_fit.x',
 'head.x',
 'hgh2qe.x',
 'hp.x',
 'ibrav2cell.x',
 'initial_state.x',
 'kpoints.x',
 'lambda.x',
 'ld1.x',
 'manycp.x',
 'matdyn.x',
 'memory_pw4gww.x',
 'molecularnexafs.x',
 'molecularpdos.x',
 'neb.x',
 'open_grid.x',
 'path_interpolation.x',
 'pawplot.x',
 'ph.x',
 'phcg.x',
 'plan_avg.x',
 'plotband.x',
 'plotproj.x',
 'plotrho.x',
 'pmw.x',
 'postahc.x',
 'postw90.x',
 'pp.x',
 'pp_disca.x',
 'pp_spctrlfn.x',
 'ppacf.x',
 'pprism.x',
 'projwfc.x',
 'pw.x',
 'pw2bgw.x',
 'pw2critic.x',
 'pw2gt.x',
 'pw2gw.x',
 'pw2wannier90.x',
 'pw4gww.x',
 'pwcond.x',
 'pwi2xsf.x',
 'q2qstar.x',
 'q2r.x',
 'scan_ibrav.x',
 'simple.x',
 'simple_bse.x',
 'simple_ip.x',
 'spectra_correction.x',
 'sumpdos.x',
 'test_qe_fftx_fft_scalar_gpu.x',
 'test_qe_fftx_fft_scatter_mod_gpu.x',
 'test_qe_fftx_fwinv_gpu.x',
 'test_qe_lax_lapack_zdotc.x',
 'turbo_davidson.x',
 'turbo_eels.x',
 'turbo_lanczos.x',
 'turbo_magnon.x',
 'turbo_spectrum.x',
 'upfconv.x',
 'virtual_v2.x',
 'w90chk2chk.x',
 'wannier90.x',
 'wannier_ham.x',
 'wannier_plot.x',
 'wfck2r.x',
 'wfdd.x',
 'xspectra.x']

In [2]: sorted(re.findall(r"Installing:\s*\$PREFIX/bin/(.*\.x)", text))
Out[2]: 
['ZG.x',
 'abcoeff_to_eps.x',
 'all_currents.x',
 'alpha2f.x',
 'bands_unfold.x',
 'bse_main.x',
 'casino2upf.x',
 'cell2ibrav.x',
 'cp.x',
 'cppp.x',
 'disca.x',
 'dvscf_q2r.x',
 'dynmat.x',
 'epa.x',
 'epsilon_Gaus.x',
 'epw.x',
 'ev.x',
 'fd.x',
 'fd_ef.x',
 'fd_ifc.x',
 'fqha.x',
 'graph.x',
 'gww.x',
 'gww_fit.x',
 'head.x',
 'hgh2qe.x',
 'hp.x',
 'ibrav2cell.x',
 'kcw.x',
 'kcwpp_interp.x',
 'kcwpp_sh.x',
 'kpoints.x',
 'lambda.x',
 'ld1.x',
 'manycp.x',
 'matdyn.x',
 'memory_pw4gww.x',
 'molecularnexafs.x',
 'neb.x',
 'path_interpolation.x',
 'ph.x',
 'phcg.x',
 'postahc.x',
 'postw90.x',
 'pp_disca.x',
 'pp_spctrlfn.x',
 'pw.x',
 'pw4gww.x',
 'pwcond.x',
 'pwi2xsf.x',
 'q2qstar.x',
 'q2r.x',
 'qe_xc_infos.x',
 'scan_ibrav.x',
 'simple.x',
 'simple_bse.x',
 'simple_ip.x',
 'spectra_correction.x',
 'turbo_davidson.x',
 'turbo_eels.x',
 'turbo_lanczos.x',
 'turbo_magnon.x',
 'turbo_spectrum.x',
 'upfconv.x',
 'virtual_v2.x',
 'w90chk2chk.x',
 'wannier90.x',
 'wfdd.x',
 'xspectra.x']
```

@ltalirz @jan-janssen  any idea?

**Comment by chrisjsewell** [comment]
@conda-forge-admin, please rerender

**Comment by ltalirz** [comment]
Thanks for fixing the build @chrisjsewell .

I think for the moment it's ok to simply update the tests to check for the executables that are currently being installed.

Perhaps @ye-luo may be able to comment on whether this reduction in executables being installed was intentional.

**Comment by ltalirz** [comment]
P.S. Here are the missing ones
``` 
['average.x', 'bands.x', 'dos.x', 'dos_sp.x', 'ef.x', 'epsilon.x', 'fermi_int_0.x', 'fermi_int_1.x', 'fermi_proj.x', 'fermi_velocity.x', 'fs.x', 'initial_state.x', 'molecularpdos.x', 'open_grid.x', 'pawplot.x', 'plan_avg.x', 'plotband.x', 'plotproj.x', 'plotrho.x', 'pmw.x', 'pp.x', 'ppacf.x', 'pprism.x', 'projwfc.x', 'pw2bgw.x', 'pw2critic.x', 'pw2gt.x', 'pw2gw.x', 'pw2wannier90.x', 'sumpdos.x', 'test_qe_fftx_fft_scalar_gpu.x', 'test_qe_fftx_fft_scatter_mod_gpu.x', 'test_qe_fftx_fwinv_gpu.x', 'test_qe_lax_lapack_zdotc.x', 'wannier_ham.x', 'wannier_plot.x', 'wfck2r.x']
``` 

There are some rather important executables in there, so that does look more like a bug

**Comment by ye-luo** [comment]
https://gitlab.com/QEF/q-e/-/merge_requests/2007
should fix all the pp targets at install. No intention to install test executables.

**Comment by chrisjsewell** [comment]
thanks @ye-luo!

I added a patch file for your change, so let's see how that goes 🤞

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+  sha256: d56dea096635808843bd5a9be2dee3d1f60407c01dbeeda03f8256a3bcfc4eb6
   patches:
-   # drop following patch for next version after 7.0
-   - 979aba59d385d4db0a485f17e3b0b3937be1a9cf.diff
+  # can be removed once https://gitlab.com/QEF/q-e/-/merge_requests/2007 is released
+  - fix_make_install.patch
 
 build:
-  number: 2
+  number: 1
```
Comment (chrisjsewell) [inline recipe/meta.yaml:16]:
```suggestion
  number: 0
```

**Comment by chrisjsewell** [comment]
it works 🎉

**Comment by chrisjsewell** [comment]
@ltalirz etc good to go?

**Review by jan-janssen — APPROVED** [review]
Looks good to me

**Comment by ltalirz** [comment]
Great, thanks a lot @chrisjsewell and @ye-luo !

**Pull request merged**